src/lib.rs

```diff
@@ -1404,8 +1404,9 @@ impl Builder {
     /// Generate the Rust bindings using the options built up thus far.
     pub fn generate(mut self) -> Result<Bindings, ()> {
         // Add any extra arguments from the environment to the clang command line.
-        if let Some(extra_clang_args) =
-            env::var("BINDGEN_EXTRA_CLANG_ARGS").ok()
+        if let Ok(extra_clang_args) = env::var(&format!("BINDGEN_EXTRA_CLANG_ARGS_{}", env::var("TARGET").unwrap()))
+            .or_else(|_| env::var(&format!("BINDGEN_EXTRA_CLANG_ARGS_{}", env::var("TARGET").unwrap().replace("-", "_"))))
+            .or_else(|_| env::var("BINDGEN_EXTRA_CLANG_ARGS"))
```

Comment:
This can't `unwrap` the `env::var()` result, because when used from the CLI it might not be set.

Reply:
Yes ! The unit tests caught it. I'm changing it to .map()

Reply:
Can we move this to a reusable function? Something like:

```rust
fn get_target_dependent_env_var(var: &str) -> Option<String> {
    if let Some(target) = env::var("TARGET") {
        if let Some(v) = env::var(&format!("{}_{}", var, target)) {
            return Some(v);
        }
        if let Some(v) = env::var(&format!("{}_{}", var, target.replace("...")) {
            return Some(v)
        }
    }
    return env::var(var);
}
```

or such?

Reply:
done !